src/controllers/userController.js

```diff
@@ -36,4 +36,34 @@ export default class UserController {
         });
     }
   }
+
+  static updateUserProfile(req, res, next) {
+    User.findById(req.payload.id)
+      .then((user) => {
+        if (!user) {
+          return res.sendStatus(401);
+        }
+
+        // only update fields that were actually passed...
+        if (typeof req.body.user.username !== 'undefined') {
+          user.username = req.body.user.username;
+        }
+        if (typeof req.body.user.email !== 'undefined') {
+          user.email = req.body.user.email;
+        }
+        if (typeof req.body.user.bio !== 'undefined') {
+          user.bio = req.body.user.bio;
+        }
+        if (typeof req.body.user.image !== 'undefined') {
+          user.image = req.body.user.image;
+        }
+        if (typeof req.body.user.password !== 'undefined') {
+          user.setPassword(req.body.user.password);
+        }
+
+        return user.save().then(() => res.json({ user: user.toAuthJSON() }));
+      })
+      .catch(next);
+  }
+
```

Comment:
We removed this function I don't know why you still have them. That's why you have many untested lines

Reply:
Okay, let me remove it also.